sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCanonicalize.scala

```diff
@@ -55,6 +53,16 @@ object GpuCanonicalize {
     case _ => e
   }
 
+  /** Remove TimeZoneId for Cast if needsTimeZone return false. */
+  private def ignoreTimeZone(e: Expression): Expression = e match {
+    case c: TimeZoneAwareExpression
+      if c.prettyName.contains("Cast") &&
```

Comment:
I don't like using `prettyName` here because it hides the fact that this is for cast. Here we could find it by using a String search, but we would not ever find it using the IDE to places `GpuCast` or `Cast` is used. I am not going to stop this going in but I am very much a -0 on this. I would rather have the shim.

Reply:
In addition to noting that we still reference the companion object Cast via Cast.needsTimeZone
will it help if we 
- change the method name for ignoreTimeZoneInCast
- add back the match for GpuCast to make it easier to locate the code in IDE?
```scala
  private def ignoreTimeZone(e: Expression): Expression = e match {
    case c: GpuCast if c.timeZoneId.nonEmpty && !c.needsTimeZone =>
      c.withTimeZone(null)
    case c: TimeZoneAwareExpression
      if c.prettyName.contains("Cast") &&
        c.timeZoneId.nonEmpty &&
        !Cast.needsTimeZone(c.children.head.dataType, c.dataType) =>
      c.withTimeZone(null)
    case _ => e
  }
```